### PR TITLE
#103 thrift server chart

### PR DESCRIPTION
**Pull request by peter-mcclonski**

### CODE REVIEW

File: foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.yaml.vm [hunk 1]
```diff
@@ -1,5 +1,42 @@
...
+      AWS_ACCESS_KEY_ID:
+        secretName: aws-s3-credentials
+        key: aws_access_key_id
+      AWS_SECRET_ACCESS_KEY:
+        secretName: aws-s3-credentials
+        key: aws_secret_access_key
+
+  sparkConf: |
+    spark.hadoop.fs.s3a.endpoint=http://s3-local:4566
+    spark.hadoop.fs.s3a.access.key=#[[${env:AWS_ACCESS_KEY_ID}]]#
```
Comment (peter-mcclonski) [inline foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.yaml.vm:23]: I: The `#[[ ... ]]#` syntax instructs Velocity not to attempt to parse the inner text.

File: foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.yaml.vm [hunk 2]
```diff
@@ -1,5 +1,42 @@
-# This file contains the default values for the aiSSEMBLE Spark History Helm chart.
+# This file contains the default values for the aiSSEMBLE Spark Infrastructure Helm chart.
 
 aissemble-spark-history-chart:
-    eventVolume:
-      enabled: true
\ No newline at end of file
+  eventVolume:
```
Comment (peter-mcclonski) [inline foundation/foundation-mda/src/main/resources/templates/deployment/spark-infrastructure/v2/spark.infrastructure.values.yaml.vm:4]: I: Fixed inconsistent indentation

**Review by cpointe-ibllanos — APPROVED** [review]
Reviewed all the files. Looks good.

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/templates/deployment.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ .Release.Name }}
+  name: {{ .Release.Name }}-shs
```
Comment (ewilkins-csi) [inline extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/templates/deployment.yaml:5]: Q: Wouldn't the release name already mention that it's a spark history server? So it'd be something like `my-spark-history-shs` which feels odd.

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+| deployment.replicas              | Number of replicas for the Spark Thrift Server Deployment.                                                                        | 1                                                                     |
+| deployment.image.repository      | Repository for the Spark Thrift Server image.                                                                                     | "apache/spark"                                                        |
+| deployment.image.tag             | Tag for the Spark Thrift Server image.                                                                                            | "3.5.1"                                                               |
+| deployment.image.imagePullPolicy | Image pull policy for the Spark Thrift Server image.                                                                              | "IfNotPresent"                                                        |
```
Comment (ewilkins-csi) [inline extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/README.md:21]: A: Why not just `deployment.image.pullPolicy` or `deployment.imagePullPolicy`?

**Review by aaron-gary — APPROVED** [review]

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/templates/deployment.yaml (hunk 3 above)
Comment (peter-mcclonski) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-spark-history-chart/templates/deployment.yaml:5]: The problem here is how it works when unified in a single chart in a downstream project.  When you generate out `spark-infrastructure` downstream, with a dependency on both the shs chart and sts chart, when the chart is deployed, `.Release.Name` is identical for both (specifically, `spark-infrastructure`)

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+          image: {{ .Values.deployment.image.repository }}:{{ .Values.deployment.image.tag }}
+          imagePullPolicy: {{ .Values.deployment.image.imagePullPolicy | default "IfNotPresent" }}
+          # Uses the Ivy jar packaged with spark to download dependency packages (and their transitive dependencies) to /tmp/jars/...
+          # Specifically downloads the binaries-- Does not download supplemental classifiers, ie sources
+          command: ["java", "-Divy.cache.dir=/tmp/", "-Divy.home=/tmp/", "-jar", "/opt/spark/jars/ivy-2.5.1.jar", "-confs", "default", "-retrieve", "/tmp/jars/[artifact].[ext]", {{ include "deps.packages.ivy" . }}]
+          volumeMounts:
+            - mountPath: /tmp/jars
+              name: sts-jars
+        {{- end }}
+        {{- if not (empty .Values.dependencies.jars) }}
```
Comment (ewilkins-csi) [inline extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/deployment.yaml:55]: Q/A: Can we not just do this with one init container with chained commands?

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            {{- if or (not (eq .Values.sparkConf "")) (not ( eq .Values.hiveSite "")) }}
+            - name: spark-config
+              mountPath: /opt/spark/conf/
+              readOnly: true
+            {{- end }}
+            {{- if .Values.deployment.volumeMounts }}
+            {{- toYaml .Values.deployment.volumeMounts }}
+            {{- end }}
+      volumes:
+        {{- if or (not (empty .Values.dependencies.packages)) (not (empty .Values.dependencies.jars)) }}
```
Comment (ewilkins-csi) [inline extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/deployment.yaml:107]: A: See this in a few places in this template. Maybe worth pulling out into a common place

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/ingress.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      http:
+        paths:
+          {{- range .paths }}
+          - path: {{ .path }}
+            pathType: {{ .pathType }}
+            backend:
+              service:
+                name: {{ .backend.service.name | default $releasename }}
+                port:
+                  number: {{ .backend.service.port.number }}
```
Comment (ewilkins-csi) [inline extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/ingress.yaml:25]: S: We know the default service name and port. We should just use the values already set on `Values.service`.  If someone actually tried to change this to a different value, I'd argue that'd be inherently incorrect.  I could see allowing them to _extend_ with extra services, but not to replace the actual service definition which we're creating in this chart.

Comment (ewilkins-csi) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/ingress.yaml:25]: Could have a static path definition for the service provided by this chart, then allow them to add extra with a `.otherPaths` config.

**Review by ewilkins-csi — APPROVED** [review]

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/README.md (hunk 4 above)
Comment (peter-mcclonski) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/README.md:21]: This is consistent with most of our other aissemble charts-- The only real departure from our normal default is storing it under the `deployment` tag as opposed to having a standalone `image` section.

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/deployment.yaml (hunk 6 above)
Comment (peter-mcclonski) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/deployment.yaml:107]: I was having issues having it work well from within `_helpers.tpl`. I'd like to take another shot at getting it to work, but don't know that it's worth holding up this ticket for.

File: extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/ingress.yaml (hunk 7 above)
Comment (peter-mcclonski) [reply extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-thrift-server-chart/templates/ingress.yaml:25]: On further examination, I would actually be inclined to take this whole section out and replace it with:

```
{{ .paths | toYaml | nindent 10 }}
```

or something to that effect

**Pull request merged**